core/lib/spree/core/controller_helpers/pricing.rb

```diff
@@ -7,12 +7,46 @@ module Pricing
         extend ActiveSupport::Concern
 
         included do
-          helper_method :current_pricing_options
+          # Key where the current currency should be stored in a requests' session
+          CURRENCY_SESSION_KEY = 'currency'
+
+          helper_method :current_pricing_options,
+                        :current_currency,
+                        :switch_currency
         end
 
+        # @see Variant::PricingOptions#from_context
         def current_pricing_options
           Spree::Config.pricing_options_class.from_context(self)
         end
+
+        # Current currency from {#current_pricing_options}
+        #
+        # @return [String]
+        # @see #current_pricing_options
+        # @example
+        #   'USD'
+        def current_currency
+          current_pricing_options.currency
```

Comment:
Question: I don't get why we can't use the value of the currency from the session here. Why the currency set on `current_pricing_options` represents the `current_currency` instead?

Reply:
When users haven't selected a currency, the session has no information about it. In this case, the default for the currency is taken. This is the logic that `current_pricing_options.currency` implements.